### PR TITLE
fix: suppress some ad events when outside of an ad break

### PR DESCRIPTION
**Pull request by andrewjl-mux**
Filter out events we do not expect to receive outside of an ad break.

### CODE REVIEW

File: library/src/main/java/com/mux/stats/sdk/muxstats/MuxStatsSdkMedia3.kt [hunk 1]
```diff
@@ -6,9 +6,19 @@ import androidx.annotation.OptIn
 import androidx.media3.common.MediaLibraryInfo
 import androidx.media3.common.Player
 import androidx.media3.common.util.UnstableApi
+import com.mux.android.util.oneOf
```
Comment (cjpillsbury) [inline library/src/main/java/com/mux/stats/sdk/muxstats/MuxStatsSdkMedia3.kt:9]: nitpick: Unused (assuming vestigial from a refactor). Remove.

File: library/src/main/java/com/mux/stats/sdk/muxstats/MuxStatsSdkMedia3.kt [hunk 2]
```diff
@@ -145,7 +155,23 @@ class AdCollector private constructor(
   }
 
   fun dispatch(event: AdEvent) {
-    eventBus.dispatch(event)
+    if (muxPlayerState == MuxPlayerState.PLAYING_ADS) {
+      eventBus.dispatch(event)
+    } else {
```
Comment (cjpillsbury) [inline library/src/main/java/com/mux/stats/sdk/muxstats/MuxStatsSdkMedia3.kt:160]: nitpick: `else if` vs. `else` + (nested) `if`?

File: library/src/test/java/com/mux/stats/sdk/muxstats/AdCollectorTests.kt [hunk 3]
```diff
@@ -45,4 +50,181 @@ class AdCollectorTests : AbsRobolectricTest() {
...
+        dispatchedEvents += (call.invocation.args.first() as IEvent)
+      }
+    }
+    val stateCollector = MuxStateCollector(mockk<MuxStats>(relaxed = true), eventBus)
+    val adCollector = AdCollector.create(stateCollector, eventBus)
+
+    adCollector.dispatch(AdPlayEvent(null))
+    adCollector.dispatch(AdPlayingEvent(null))
+
+    Assert.assertTrue(
```
Comment (cjpillsbury) [inline library/src/test/java/com/mux/stats/sdk/muxstats/AdCollectorTests.kt:71]: suggestion(non-blocking): use a loop + all expected filtered out ad events for your assertions. Alternatively, just write individual tests for every ad event like you did for the "allowed" ones, below (which allows test to better isolate an issue)

**Review by cjpillsbury — CHANGES_REQUESTED** [review]
On a quick glance only nits and minor suggestions. Otherwise, LGTM!

File: library/src/main/java/com/mux/stats/sdk/muxstats/MuxStatsSdkMedia3.kt (hunk 1 above)
Comment (andrewjl-mux) [reply library/src/main/java/com/mux/stats/sdk/muxstats/MuxStatsSdkMedia3.kt]: c1899b3d97eb402767ee2889da25fdf425e75da2

File: library/src/main/java/com/mux/stats/sdk/muxstats/MuxStatsSdkMedia3.kt (hunk 2 above)
Comment (andrewjl-mux) [reply library/src/main/java/com/mux/stats/sdk/muxstats/MuxStatsSdkMedia3.kt]: 760e7d6ef45e0f33d7f77dcfea449f66a5a44e4b

**Review by cjpillsbury — APPROVED** [review]
👍

**Review by daytime-em — APPROVED** [review]
me three

**Pull request merged**